server/src/main/java/org/apache/iotdb/db/query/reader/universal/PriorityMergeReader.java

```diff
@@ -23,143 +23,137 @@
-//  public TimeValuePair next() throws IOException {
-//    Element top = heap.poll();
-//    TimeValuePair ret = top.timeValuePair;
-//    TimeValuePair topNext = null;
-//    if (top.hasNext()) {
-//      top.next();
-//      topNext = top.currPair();
-//    }
-//    long topNextTime = topNext == null ? Long.MAX_VALUE : topNext.getTimestamp();
-//    updateHeap(ret.getTimestamp(), topNextTime);
-//    if (topNext != null) {
-//      top.timeValuePair = topNext;
-//      heap.add(top);
-//    }
-//    return ret;
-//  }
-//
-//  @Override
-//  public TimeValuePair current() throws IOException {
-//    return heap.peek().timeValuePair;
-//  }
-//
-//  private void updateHeap(long topTime, long topNextTime) throws IOException {
-//    while (!heap.isEmpty() && heap.peek().currTime() == topTime) {
-//      Element e = heap.poll();
-//      if (!e.hasNext()) {
-//        e.reader.close();
-//        continue;
-//      }
-//
-//      e.next();
-//      if (e.currTime() == topNextTime) {
-//        // if the next value of the peek will be overwritten by the next of the top, skip it
-//        if (e.hasNext()) {
-//          e.next();
-//          heap.add(e);
-//        } else {
-//          // the chunk is end
-//          e.close();
-//        }
-//      } else {
-//        heap.add(e);
-//      }
-//    }
-//  }
-
-  @Override
-  public boolean hasNextBatch() throws IOException {
-    return false;
+  public boolean hasNext() {
+    return !heap.isEmpty();
   }
 
-  @Override
-  public BatchData nextBatch() throws IOException {
-    return null;
+  public Element next() throws IOException {
+    Element top = heap.poll();
+    long ret = top.time;
+    long topNextTime = Long.MAX_VALUE;
+    Object topNextValue = null;
+    if (top.hasNext()) {
+      top.next();
+      topNextTime = top.currTime();
+      topNextValue = top.currValue();
+    }
+    updateHeap(ret, topNextTime);
+    if (topNextValue != null) {
+      top.time = topNextTime;
+      top.value = topNextValue;
+      heap.add(top);
+    }
+    return top;
+  }
+
+  public Element current() throws IOException {
+    return heap.peek();
+  }
+
+  private void updateHeap(long topTime, long topNextTime) throws IOException {
+    while (!heap.isEmpty() && heap.peek().currTime() == topTime) {
+      Element e = heap.poll();
+      if (!e.hasNext()) {
+        e.reader.close();
+        continue;
+      }
+
+      e.next();
+      if (e.currTime() == topNextTime) {
+        // if the next value of the peek will be overwritten by the next of the top, skip it
+        if (e.hasNext()) {
+          e.next();
+          heap.add(e);
+        } else {
+          // the chunk is end
+          e.close();
+        }
+      } else {
+        heap.add(e);
+      }
+    }
   }
 
-  @Override
   public void close() throws IOException {
     while (!heap.isEmpty()) {
       Element e = heap.poll();
       e.close();
     }
   }
 
-  class Element {
+  public class Element {
 
     IPointReader reader;
```

Comment:
From my current understanding, there is a future TODO:
 `ChunkReader` is going to implement only the `IBatchReader` operator. In other words, there will be no more `IPointReader` or related methods such as ` TimeValuePair timeValuePair = reader.next();`.

Reply:
The future TODO is out of the scope of this pr. This pr is good.

Reply:
Yes, I intend to make `DiskChunkReader` and `MemChunkReader` only implement `IBatchReader` in next step.